src/client.ts

```diff
@@ -231,6 +233,15 @@ export class SkynetClient {
 
     const auth = config.APIKey ? { username: "", password: config.APIKey } : undefined;
 
+    const onDownloadProgress =
+      config.onDownloadProgress &&
+      /* istanbul ignore next */
+      function (event: ProgressEvent) {
+        const progress = event.loaded / event.total;
+
+        // Need the if-statement or TS complains.
+        if (config.onDownloadProgress) config.onDownloadProgress(progress, event);
```

Comment:
`nit` I'm not a fan of inline `if`s - should always write them in full imo with the curly brackets.